### PR TITLE
DOC: Add linspace range behavior to the timeseries/timedeltas/interval docs

### PR DESCRIPTION
**Pull request by jschendel**
Follow-up to #21009

`timeseries.rst`:
- Added documentation for linspace behavior of `date_range`

`timedeltas.rst`:
- Cleaned up `timedelta_range` documentation for the standard behavior
- Added documentation for linspace behavior of `timedelta_range`

`advanced.rst`:
- Added documentation for the standard behavior of `interval_range`
- Added documentation for linspace behavior of `interval_range`

### CODE REVIEW

File: doc/source/timedeltas.rst [hunk 1]
```diff
@@ -363,13 +363,44 @@ or ``np.timedelta64`` objects. Passing ``np.nan/pd.NaT/nat`` will represent miss
...
+   pd.timedelta_range(start='1 days', periods=5)
+
+Various combinations of ``start``, ``end``, and ``periods`` can be used with
+``timedelta_range``:
+
+.. ipython:: python
+
+   pd.timedelta_range(start='1 days', end='5 days')
+
+   pd.timedelta_range( end='10 days', periods=4)
```
Comment (mroeschke) [inline doc/source/timedeltas.rst]: Small nit: There's an extra space between the left parenthesis and end `pd.timedelta_range( end=...)`

**Comment by mroeschke** [comment]
Small comment otherwise LGTM.

File: doc/source/timedeltas.rst (hunk 1 above)
Comment (jschendel) [reply doc/source/timedeltas.rst]: thanks, done

File: doc/source/advanced.rst [hunk 2]
```diff
@@ -924,6 +924,51 @@ bins, with ``NaN`` representing a missing value similar to other dtypes.
 
    pd.cut([0, 3, 5, 1], bins=c.categories)
 
+If we need intervals on a regular frequency, we can use the :func:`interval_range` function
```
Comment (jreback) [inline doc/source/advanced.rst]: can you add a sub-section here

File: doc/source/timedeltas.rst [hunk 3]
```diff
@@ -363,13 +363,44 @@ or ``np.timedelta64`` objects. Passing ``np.nan/pd.NaT/nat`` will represent miss
    pd.TimedeltaIndex(['1 days', '1 days, 00:00:05',
                      np.timedelta64(2,'D'), datetime.timedelta(days=2,seconds=2)])
 
-Similarly to ``date_range``, you can construct regular ranges of a ``TimedeltaIndex``:
+Similar to ``date_range``, you can construct regular ranges of a ``TimedeltaIndex``
```
Comment (jreback) [inline doc/source/timedeltas.rst]: can you add ``:func:`` to these

File: doc/source/timedeltas.rst [hunk 4]
```diff
@@ -363,13 +363,44 @@ or ``np.timedelta64`` objects. Passing ``np.nan/pd.NaT/nat`` will represent miss
    pd.TimedeltaIndex(['1 days', '1 days, 00:00:05',
                      np.timedelta64(2,'D'), datetime.timedelta(days=2,seconds=2)])
 
-Similarly to ``date_range``, you can construct regular ranges of a ``TimedeltaIndex``:
```
Comment (jreback) [inline doc/source/timedeltas.rst]: maybe make a sub-section here

**Review by jreback — CHANGES_REQUESTED** [review]
looks good. just some minor comments. merge when looking good.

**Pull request merged**